pkg/controller/registry/resolver/operators.go

```diff
@@ -263,7 +264,11 @@ func NewOperatorFromBundle(bundle *api.Bundle, startingCSV string, sourceKey Cat
 	if len(required) == 0 && len(provided) == 0 {
 		// fallback to csv parsing
 		if bundle.CsvJson == "" {
-			return nil, fmt.Errorf("couldn't parse bundle")
+			if bundle.GetBundlePath() != "" {
```

Comment:
this doesn't seem like the right place - if we're falling back to CSV parsing, we don't care if there's a bundlepath at all.

Reply:
I'll admit it seems odd, but I think we still want to throw a specific error when a bundlepath isn't accompanied by the expected metadata. WDYT?